src/chttpd/src/chttpd_auth_cache.erl

```diff
@@ -160,9 +160,12 @@ changes_callback({change, {Change}}, _) ->
         <<"_design/", _/binary>> ->
             ok;
         DocId ->
-            UserName = username(DocId),
-            couch_log:debug("Invalidating cached credentials for ~s", [UserName]),
-            ets_lru:remove(?CACHE, UserName)
+            try
+                UserName = username(DocId),
+                couch_log:debug("Invalidating cached credentials for ~s", [UserName]),
+                ets_lru:remove(?CACHE, UserName)
```

Comment:
Isn't it a problem that we don't remove the doc from the cache if it has an invalid name? It seems like it should be removed if it's stale, despite it having an unexpected prefix.

Is it possible (or desirable) to prevent such a doc from even entering the cache at all?

Reply:
I'll look into this this weekend. I would be indeed better to simply prevent the doc to enter the cache. Altought, if the user document is invalid, can we still use the user credentials to login? That's something I will check today.